src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java

```diff
@@ -56,18 +58,30 @@ public AttachCaseCallbackService(CcdApi ccdApi) {
      * @return Either a map of fields that should be modified in CCD when processing was successful,
      *         or the list of errors, in case of errors
      */
-    public Either<List<String>, Map<String, Object>> process(CaseDetails exceptionRecord) {
+    public Either<List<String>, Map<String, Object>> process(CaseDetails exceptionRecord, String eventId) {
         boolean useSearchCaseReference = isSearchCaseReferenceTypePresent(exceptionRecord);
 
-        return getValidation(exceptionRecord, useSearchCaseReference)
+        return getValidation(exceptionRecord, useSearchCaseReference, eventId)
             .map(this::tryAttachToCase)
             .getOrElseGet(errors -> Either.left(errors.toJavaList()));
     }
 
     private Validation<Seq<String>, AttachToCaseEventData> getValidation(
         CaseDetails exceptionRecord,
-        boolean useSearchCaseReference
+        boolean useSearchCaseReference,
+        String eventId
     ) {
+        // check if event id - classification is valid or not
+        Validation<String, Void> eventIdClassificationValidation =
+            hasValidCombinationOfEventIdAndClassification(exceptionRecord, eventId);
+
+        if (eventIdClassificationValidation.isInvalid()) {
+            String eventIdClassificationValidationError = eventIdClassificationValidation.getError();
+            log.warn("Validation error {}", eventIdClassificationValidationError);
+            final Seq<String> errors = io.vavr.collection.List.of(eventIdClassificationValidationError);
+            return invalid(errors);
+        }
```

Comment:
I thought we are collecting **all** the errors/warnings

Reply:
I think this is the first validation which we need to do to check if the request is eligible for processing or not. Failing fast would be best in this case as action is invalid and there is nothing much achieved by doing other valdiation.
L

Reply:
fair point. in that case, this can be done straight away first thing in the service entry, right before `boolean useSearchCaseReference = isSearchCaseReferenceTypePresent(exceptionRecord);`